flix-core/src/test/java/com/epam/mentoring/flixcore/service/MovieServiceTest.java

```diff
@@ -0,0 +1,84 @@
+package com.epam.mentoring.flixcore.service;
+
+import com.epam.mentoring.flixcore.FlixCoreApplication;
+import com.epam.mentoring.flixcore.model.Genre;
+import com.epam.mentoring.flixcore.model.Movie;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = FlixCoreApplication.class)
+public class MovieServiceTest {
+
+    @Autowired
+    private MovieService movieService;
+
+    private Movie lordOfTheRings;
+    private Movie titanic;
+
+    @Before
+    public void setUp() throws Exception {
+        lordOfTheRings = new Movie("Lord of the Rings", Genre.FANTASY, 146);
+        titanic = new Movie ( "Titanic", Genre.DRAMA, 152);
+        movieService.createMovie(lordOfTheRings);
```

Comment:
In general its a good idea to load some data into database in @Before method, but only the data that is actually needed for actual set up. For example if you would be testing some class that need to have pre existing data its fine to load it here.
**You should never sacrifice clarity to reduce cross-test duplication.**
What I would suggest is not to create movies here, but create them in particular test (specially that in most cases you dont need exactly 2 of them).

Also its really dangerous to let tests mutate data in between tests (here it wont happen as you are deleting movies in @after, but its always good to keep it in mind.

If this sounds not clear let me know and we may discuss.

Reply:
I've changed implementation a bit.
So, I moved creation into each test. Bud, still I left **@after** method where I have to remove movies from DB. Is it a good approach or we have to remove movies from DB at the end of each test as well?